File: nodejs/muse/algorithms/sorting/QuickSortTest.js
"use strict"

const QuickSort       = require("muse/algorithms/sorting/QuickSort")
const SequenceBuilder = require("muse/util/SequenceBuilder")
const Sequences       = require("muse/util/Sequences")
const TestRunner      = require("muse/util/TestRunner")

const testQuickSort = () => {
    let size = 32768

    const arr = new Array(size)
    SequenceBuilder.packRandom(arr)

    let checksum = Sequences.parityChecksum(arr)

    QuickSort.sort(arr)

    if (Sequences.parityChecksum(arr) !== checksum) {
        return false
    }

    return Sequences.isSorted(arr)
}

if (module === require.main) {
    TestRunner.parseTest(testQuickSort())
}

File: java/muse/algorithms/sorting/QuickSortTest.java
import muse.algorithms.sorting.QuickSort;
import muse.util.SequenceBuilder;
import muse.util.Sequences;
import muse.util.TestRunner;

public class QuickSortTest {

    public static boolean testQuickSort() {
        int size = 32768;

        Integer[] arr = new Integer[size];
        SequenceBuilder.packRandom(arr);

        int checksum = Sequences.parityChecksum(arr);

        QuickSort.sort(arr);

        if (Sequences.parityChecksum(arr) != checksum) {
            return false;
        }

        return Sequences.isSorted(arr);
    }

    public static void main(String[] args) {
        TestRunner.parseTest(testQuickSort());
    }
}
